packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js

```diff
@@ -87,26 +124,126 @@ export default class QuanticSearchBoxSuggestionsList extends LightningElement {
+        ...option,
+        key: index + 1,
+        isSelected: this.selectionIndex === index + 1,
+        containerCSSClass: `${optionCSSClass} ${
+          this.selectionIndex === index + 1 ? 'slds-has-focus' : ''
+        }`,
+        icon: option.isRecentQuery ? 'utility:clock' : 'utility:search',
+        clickHandler: () => {
+          this.handleSuggestionSelection(option.rawValue, option.isRecentQuery);
+        },
+      }));
+      return [
+        {
+          key: 0,
+          isSelected: this.selectionIndex === 0,
+          isClearRecentQueryButton: true,
+        },
+        ...options,
+      ].slice(0, this.maxNumberOfSuggestions + 1);
+    }
+    return this.querySuggestionsNotInRecentQueries
+      .map((option, index) => ({
+        ...option,
+        key: index,
+        isSelected: this.selectionIndex === index,
+        containerCSSClass: `${optionCSSClass} ${
+          this.selectionIndex === index ? 'slds-has-focus' : ''
+        }`,
+        icon: 'utility:search',
+        clickHandler: () => {
+          this.handleSuggestionSelection(option.rawValue);
+        },
+      }))
+      .slice(0, this.maxNumberOfSuggestions);
   }
 
-  emitSuggestionHighlighted() {
-    const highlightChangeEvent = new CustomEvent('highlightchange', {
-      detail: this.suggestions[this.selectionIndex],
+  /**
+   * Returns the query suggestions that are not already in the recent queries list.
+   */
+  get querySuggestionsNotInRecentQueries() {
+    return (
+      this.suggestions?.filter(
+        (suggestion) =>
+          !this.recentQueriesThatStartWithCurrentQuery.some(
+            (recentQuery) => recentQuery.rawValue === suggestion.rawValue
+          )
+      ) || []
+    );
+  }
+
+  /**
+   * Returns the recent queries that start with the query currently typed by the end user.
+   */
+  get recentQueriesThatStartWithCurrentQuery() {
+    return (
+      this.recentQueries
+        ?.filter(
+          (recentQuery) =>
+            recentQuery !== this.query &&
+            recentQuery.toLowerCase().startsWith(this.query?.toLowerCase())
+        )
+        .map((recentQuery) => ({
+          value: this.formatRecentQuery(recentQuery),
+          rawValue: recentQuery,
+          isRecentQuery: true,
+        })) || []
+    );
+  }
+
+  /**
+   * Highlights a recent query based on the letters that match the current query.
+   * @param {String} value
+   * @returns {String}
+   */
+  formatRecentQuery(value) {
+    const highlightedValue = CoveoHeadless.HighlightUtils.highlightString({
+      content: value,
+      openingDelimiter: '<b>',
+      closingDelimiter: '</b>',
+      highlights: [
+        {
+          offset: this.query.length,
+          length: value.length - this.query.length,
+        },
+      ],
     });
-    this.dispatchEvent(highlightChangeEvent);
+    return highlightedValue;
+  }
+
+  emitSuggestionHighlighted() {
+    if (!(this.shouldDisplayRecentQueries && this.selectionIndex === 0)) {
```

Comment:
You see this is where you're moving complexity because you have to account for the fact that the "clear recent queries" is counted in the indices of recent queries.

Would it be acceptable in terms of a11y perspective to have the "clear" be the last element instead of the first? But displayed as the first maybe?